### PR TITLE
bpo-41867: Include options for timespec in docstrings of isoformat methods

### PR DESCRIPTION
**Pull request by cool-RR**
https://bugs.python.org/issue41867

### CODE REVIEW

**Comment by taleinat** [comment]
Hi Ram, thanks for this, it's a nice addition!

What about the doc-string for the C implementation? (e.g. [here](https://github.com/python/cpython/blob/20ce62f00957d11f24f6449cd5c0ef5dd67174d4/Modules/_datetimemodule.c#L4664))

**Comment by taleinat** [comment]
> What about the doc-strings for the C implementation? (e.g. here)

It seems that the C implementation for `datetime.iso_format()` already has something very similar in its doc-string, while `time.iso_format()` doesn't.

Let's make these all consistent.

**Comment by cool-RR** [comment]
Done.

File: Misc/NEWS.d/next/Library/2020-09-26-19-45-06.bpo-41867.hcueGL.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Include the different options for `timespec` in docstrings of `isoformat`
+methods.
```
Comment (taleinat) [inline Misc/NEWS.d/next/Library/2020-09-26-19-45-06.bpo-41867.hcueGL.rst]: Please remove this, as a NEWS entry isn't necessary when only making doc-string improvements.

File: Lib/datetime.py [hunk 2]
```diff
@@ -1421,7 +1421,8 @@ def isoformat(self, timespec='auto'):
         part is omitted if self.microsecond == 0.
 
         The optional argument timespec specifies the number of additional
-        terms of the time to include.
+        terms of the time to include. Options are 'auto', 'hours', 'minutes',
```
Comment (taleinat) [inline Lib/datetime.py]: "Options are ..." reads a bit awkwardly. Perhaps "Valid options are..."? Or "Valid values", or "Possible values".

**Review by taleinat — CHANGES_REQUESTED** [review]
Almost good to go, just a couple of small comments.

File: Misc/NEWS.d/next/Library/2020-09-26-19-45-06.bpo-41867.hcueGL.rst (hunk 1 above)
Comment (cool-RR) [reply Misc/NEWS.d/next/Library/2020-09-26-19-45-06.bpo-41867.hcueGL.rst]: Done.

File: Lib/datetime.py (hunk 2 above)
Comment (cool-RR) [reply Lib/datetime.py]: Done.

**Comment by cool-RR** [comment]
I have made the requested changes; please review again

File: Modules/_datetimemodule.c [hunk 3]
```diff
@@ -4663,7 +4663,10 @@ static PyMethodDef time_methods[] = {
     {"isoformat",   (PyCFunction)(void(*)(void))time_isoformat,        METH_VARARGS | METH_KEYWORDS,
      PyDoc_STR("Return string in ISO 8601 format, [HH[:MM[:SS[.mmm[uuu]]]]]"
                "[+HH:MM].\n\n"
-               "timespec specifies what components of the time to include.\n")},
+               "The optional argument timespec specifies the number of "
+               "additional terms of the time to include. Valid options are "
+               "'auto', 'hours', 'minutes', 'seconds', 'milliseconds' and "
+               "'microseconds'.\n")},
```
Comment (taleinat) [inline Modules/_datetimemodule.c]: This is one very long doc-string line, which needs to be broken up with `\n`.

(below as well)

**Review by taleinat — CHANGES_REQUESTED** [review]
Wow, @cool-RR, that was quick!

One more thing...

**Comment by cool-RR** [comment]
I have made the requested changes; please review again

**Comment by cool-RR** [comment]
We pride ourselves on fast service :)

I do wish the text formatting inside that C file wasn't so horrible. I don't think programmers should spend so much of their time juggling words and `\n` escape sequences inside strings to do string formatting.

File: Modules/_datetimemodule.c [hunk 4]
```diff
@@ -4663,7 +4663,10 @@ static PyMethodDef time_methods[] = {
     {"isoformat",   (PyCFunction)(void(*)(void))time_isoformat,        METH_VARARGS | METH_KEYWORDS,
      PyDoc_STR("Return string in ISO 8601 format, [HH[:MM[:SS[.mmm[uuu]]]]]"
                "[+HH:MM].\n\n"
-               "timespec specifies what components of the time to include.\n")},
+               "The optional argument timespec specifies the number of "
+               "additional terms of the time to include.\n"
```
Comment (taleinat) [inline Modules/_datetimemodule.c]: Sorry, this is still too long of a line at 95 characters... :(

Doc-sting lines should be wrapped to 72 chars.

(Yes, I realize that this is rather annoying to get right.)

**Review by taleinat — CHANGES_REQUESTED** [review]

File: Modules/_datetimemodule.c (hunk 4 above)
Comment (cool-RR) [reply Modules/_datetimemodule.c]: Done. A bit of a double standard with the first line of that docstring...

**Comment by cool-RR** [comment]
I have made the requested changes; please review again

**Review by taleinat — APPROVED** [review]
LGTM

**Pull request merged**